### DISCUSSION TITLE
Sentry-native required debug information is missing

### DISCUSSION THREAD

**Discussion by martirosyanH** [getsentry/sentry-native#850] [Q&A] [answered]
Hi everyone,

I set up sentry-native for my application. I have a test crash, and after that, the crash report is sent to the portal with a mini-dump.
For 64 bit version,  in the portal, I can see stack information(since I uploaded .exe and .pdb), but for 32 bit as the documentation says, exe files rarely thay contain unwind information, so I uploaded only pdb file for 32 bit version.

For 32-bit in the portal I can see :

There was 1 problem processing this event:
A required debug information file was missing. File Name: yyyy.exe

Any help would be appreciated.
Thanks in advance.

**Comment by supervacuus** [comment]
Hi @martirosyanH, I cannot comment on the exact behavior you see because what is required as debug information is a decision made by the backend processing. Have you tried to upload the 32-bit executable nonetheless? I wouldn't discriminate between 32/64 PE and always upload the PE and the PDB. 

Other useful information may be gathered from the 32-bit PE which hasn't found its way back into the docs, or the mechanism checking for required files doesn't discriminate (only when actual parsing for symbols, unwind-, and debug info happens). @Swatinem?

**Reply by martirosyanH** [reply]
Hi @supervacuus ,
Thanks for the quick response.

Of course I tried to upload 32 bit exe as well, but I get an “No debug information found “ message from cli, so for 32 bit I uploaded only pdb, but seems for call stack parsing exe is needed as well.
Just to note, locally I can parse the generated dump file with the 32bit exe and pdb. So there is no problem with my pdb files


For64 bit everything works, since I can upload both exe and pdb.

**Reply by tian-yu-ocado** [reply]
Hi @supervacuus, Sorry to bump up this question again, just wondering is there a way to send the debug info or stack traces as part of the crash event, instead of having to upload the debug symbols separately?

**Reply by supervacuus** [reply]
There is currently no way to upload debug info from the Native SDK. Stack traces will be sent with the minidump during the crash event.

The reasoning for the former is:
* in almost all deployment scenarios, there will be no debug info available to the client (which you might need not only for your executable or library but also for your dependencies if the bug involves frames that reference those)
* the upload of debug-info requires authentication (which you typically do not want to distribute with your deployment artifact)
* This is why uploading debug symbols is typically part of the build side, which happens in a controlled environment and is thus easy to automate as a post-build step.

Of course, I can imagine counter use-cases where these points do not apply, but I am unsure whether such use cases would lead to prioritizing such a feature.

**Comment by Swatinem** [comment, marked as answer]
You should indeed upload both exe and pdb in both cases. If you get a "no debug info found" message and therefore the file is rejected, that does look like a bug that needs to be fixed.

Otherwise the Windows 32-bit unwinding support is a bit sub-par, so its quite possible that we won’t do a good job unwinding the minidump in that case anyways.
